src/Microsoft.OData.Core/Json/DefaultStreamBasedJsonWriterFactory.cs

```diff
@@ -50,6 +50,11 @@ public IJsonWriter CreateJsonWriter(Stream stream, bool isIeee754Compatible, Enc
 
             return new ODataUtf8JsonWriter(stream, isIeee754Compatible, encoding, encoder: this.encoder);
         }
+
+        public IJsonWriterAsync CreateAsynchronousJsonWriter(Stream stream, bool isIeee754Compatible, Encoding encoding)
+        {
+            throw new NotImplementedException();

```

Comment:
Why are we throwing `NotImplementedException`?

Reply:
@KenitoInc The explanation is provided in the description

Reply:
Aah, got it! I just looked at the pending PR.